### ISSUE TITLE
any pwa example?

### ISSUE THREAD

**Issue by hiqsociety** [DioxusLabs/dioxus#414]
any pwa example?

**Comment by jkelleyrtp** [comment]
Not yet - but we will be dogfooding PWA support with an overhaul of dioxuslabs.com soon, so there will be one there :)

**Comment by arqalite** [comment]
@jkelleyrtp - Does Dioxus actually have any features specific to PWAs that we should use?

I'm working on a Dioxus Web app right now, and I started work on making it a PWA. So far it's installable, but it doesn't have any offline functionality (my service worker is empty right now - I assume caching the files Trunk generates should be enough to have it run offline).

To make it installable I only had to add a manifest and the empty service worker, if that's all that needed I can whip up a basic example (and I'd probably write the service worker too). However if Dioxus has some related stuff, let me know so I can try them out.

**Comment by jkelleyrtp** [comment]
> @jkelleyrtp - Does Dioxus actually have any features specific to PWAs that we should use?
> 
> I'm working on a Dioxus Web app right now, and I started work on making it a PWA. So far it's installable, but it doesn't have any offline functionality (my service worker is empty right now - I assume caching the files Trunk generates should be enough to have it run offline). I'm also going to work on using localStorage to save data, I assume Gloo will do the job well.
> 
> To make my app installable I only had to add a manifest and the empty service worker, if that's all that needed I can whip up a basic example (and I'd probably write the service worker too). However if Dioxus has some related stuff, let me know so I can try them out.

I don't think there's anything specific that Dioxus needs to do to enable PWA support. We just render. You'll need the service worker like you've seen.

The blocker for releasing any PWA example has been support for workers in the dioxus-cli, but I'd be more than happy to showcase an example that uses trunk for a PWA.

**Comment by mobilelifeful** [comment]
Any update, please. Wonder where we can learn more about dioxus-pwa.

**Comment by andreyddk** [comment]
Same here, awaiting pwa 🙂

**Comment by jkelleyrtp** [comment]
Working on a PWA right now - looks like we might want to add some web worker / service worker support to the CLI.

**Comment by arqalite** [comment]
Hey - I finished that app I was talking about a few months back. It could be useful for inspiration, although I don't think I made very good decisions while developing it.
https://github.com/arqalite/rummy-nights

I've had a lot of trouble with Dioxus CLI so I used Trunk instead. I have a basic service worker with installability and offline functionality going on, so no crazy web APIs or anything. If anyone needs help setting up their Dioxus project to make it a PWA, reach out and I'll see what I can do.

Basically you just need a service worker and a manifest - then throw your app into PWABuilder and it'll tell you what other things you need.

**Comment by arqalite** [comment]
I created a barebones PWA example with Trunk - this should be pretty much a working template for a PWA project: https://github.com/arqalite/dioxus-trunk-pwa

**Comment by GoodWork0903** [comment]
Hi everyone,
we can't make Dioxus PWA using dioxus-cli?

**Comment by stefnotch** [comment]
@GoodWork0903 There is an example at https://github.com/DioxusLabs/dioxus/tree/main/examples/10-integrations/pwa
